test/core-test/time_evolution.jl

```diff
@@ -93,6 +97,8 @@
         sol_me_string = sprint((t, s) -> show(t, "text/plain", s), sol_me)
         sol_mc_string = sprint((t, s) -> show(t, "text/plain", s), sol_mc)
         sol_sse_string = sprint((t, s) -> show(t, "text/plain", s), sol_sse)
+        @test prob_me.f.f isa MatrixOperator
+        @test prob_mc.f.f isa ScaledOperator # TODO: can be optimized as MatrixOperator
```

Comment:
@albertomercurio 
I add a TODO here.
cause I tried to change `_mcsolve_make_Heff_QobjEvo` to return `QuantumObject` only for time-independent case (since `sesolveProblem` will multiply `-1im`). 

I thought the `QobjEvo` can be created inside `sesolveProblem`.
But, at the end, it breaks many tests in dynamical solvers. So I leave it as `ScaledOperator` here.

If you have time, maybe you can fix this in the next PR.